### DISCUSSION TITLE
Using Quarkus for Raw TCP Sockets

### DISCUSSION THREAD

**Discussion by kalvinnchau** [quarkusio/quarkus#25609] [Q&A]
Hi,

Are there any examples of using Quarkus to open up a plain TCP (potentially UDP) socket that'll accept data do some transformations and pass data along the event bus? Most of the examples I see revolve around setting up a REST endpoint and going from there. 
Is that the correct usage of Quarkus or is that a non-goal of this project?

Thanks

**Comment by millerjp** [comment]
Something equivalent to this would be useful https://docs.spring.io/spring-integration/reference/html/ip.html

**Comment by mschorsch** [comment]
Yes this is possible. You have to inject a [Vertx-Instance](https://quarkus.io/guides/vertx-reference#vertx-access)  and create a [NetServer](https://vertx.io/docs/vertx-core/java/#_writing_tcp_servers_and_clients) or a [DatagrammSocket](https://vertx.io/docs/vertx-core/java/#_datagram_sockets_udp).

- https://quarkus.io/guides/vertx-reference#vertx-access
- https://quarkus.io/guides/vertx#using-the-event-bus
- https://vertx.io/docs/vertx-core/java/#_writing_tcp_servers_and_clients
- https://vertx.io/docs/vertx-core/java/#_datagram_sockets_udp
